src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,668 @@
+        If all is well the following variables are set, and a few others on which they (especially self.session)
+        depend.
+        self.initialised : Static variable to say if the REST interface is available and ready
+        self.base_url : The basic URL for the REST interface
+        self.session  : The python "requests" module session
+        self.headers  : The format of information we prefer : json of course (except during delegations)
+        Specification : https://www.nordugrid.org/arc/arc6/tech/rest/rest.html
+        """
+        if hasattr(self, "s"):
+            # Initialisation has already been successfully ran
+            return True
+        self.log.debug("Testing if the REST interface is available", "for %s" % self.ceHost)
+
+        result = self._prepareProxy()
+        if not result["OK"]:  # Probably everything is going to fail?
+            self.log.info("No proxy found -  Probably just no jobs to run for this CE?")
+            self.log.info("Setting REST interface false.")
+            return False
+
+        # Get the REST endpoint (service_url in ARC language) from the CS if available (preferred).
+        # It should be at
+        #   Resources/Sites/<Grid>/<Site>/CEs/<CE>/RESTEndpoint
+        # Otherwise query the CE for this value.
+
+        # Try getting service_url from the CS
+        ceString = cfgPath("Resources", "Sites", self.grid, self.site, "CEs", self.ceHost, "RESTEndpoint")
+        result = self._getFromCfgSvc(ceString)
+        if result["OK"]:
+            service_url = result["Value"]  # All well
+        if len(service_url) < 5:  # There is no endpoint in the CS. Discover it.
+            # The following command should expand to for example
+            # ldapsearch -x -LLL -h grendel.hec.lancs.ac.uk:2135 -b 'o=glue' GLUE2EndpointInterfaceName=org.nordugrid.arcrest GLUE2EndpointURL | grep GLUE2EndpointURL | awk -F ": " '{print $2}'
+            server = ldap3.Server("ldap://" + self.ceHost + ":2135")
+            try:
+                connection = ldap3.Connection(server, client_strategy=ldap3.SAFE_SYNC, auto_bind=True)
+            except ldap3.core.exceptions.LDAPSocketOpenError:
+                self.log.error("Could not connect to server", "CE : %s:2135" % self.ceHost)
+                return False
+            status, result, response, _ = connection.search(
+                "o=glue", "(GLUE2EndpointInterfaceName=org.nordugrid.arcrest)", attributes="GLUE2EndpointURL"
+            )
+            if not status or len(response) == 0:
+                self.log.error("Bad status from LDAP search", result)
+                return False
+            if len(response) != 1:
+                self.log.warning("Expected one endpoint, got %s. Using the first." % len(response))
+            # For some reason the response is of class "bytes"
+            service_url = response[0]["raw_attributes"]["GLUE2EndpointURL"][0].decode()
```

Comment:
```suggestion
            attributes = response[0]["raw_attributes"]  # pylint: disable=unsubscriptable-object
            service_url = attributes["GLUE2EndpointURL"][0].decode()
```

Reply:
Yes - this is kind of the solution I went for.